packages/babel-helper-member-expression-to-functions/src/index.js

```diff
@@ -37,9 +37,113 @@ const handle = {
   handle(member) {
     const { node, parent, parentPath } = member;
 
+    if (member.isOptionalMemberExpression()) {
+      const root = member.find(({ node, parent, parentPath }) => {
+        if (parentPath.isOptionalMemberExpression()) {
+          return parent.optional || parent.object !== node;
+        }
+        if (parentPath.isOptionalCallExpression()) {
+          return parent.optional || parent.callee !== node;
+        }
+        return true;
+      });
+
+      const rootParentPath = root.parentPath;
+      if (
+        rootParentPath.isUpdateExpression({ argument: node }) ||
+        rootParentPath.isAssignmentExpression({ left: node })
+      ) {
+        throw member.buildCodeFrameError(`can't handle assignment`);
+      }
+      if (rootParentPath.isUnaryExpression({ operator: "delete" })) {
+        throw member.buildCodeFrameError(`can't handle delete`);
```

Comment:
Those two errors are also handled by the parser, right?

Reply:
`delete foo.#bar` is handled by parser, but it's technically legal for `delete super.foo` which is also handled by this. I had written member-expression-to-functions to be generic enough to use on any kind of AST, so I added the error cases here.

I also plan to propose `foo?.bar = 1` eventually, it's just not included the original optional chaining proposal. So eventually, we'd need to add this.

Reply:
The current implementation also disallows `foo?.#bar.qux`, which is allowed in the spec. That said I am happy to fix this issue in patch releases before it gets too big.